src/build_system.ml

```diff
@@ -506,17 +506,6 @@ let get_dir_status t ~dir =
           }
     end)
 
-let entry_point t ~f =
```

Comment:
Something must have gone wrong here, `entry_point` was supposed to be called for every entry point such as `do_build`. It doesn't make sense that it is only used for `evaluate_rules`. Anyway, this is not specific to this PR but we should look into it at some point